test/test-parse.cpp

```diff
@@ -539,6 +539,111 @@ TEST(test_escape_first_character_in_keyword,
   EXPECT_EQ(escape_first_character_in_keyword(u8"b1n z"_sv), u8"\\u{62}1n z");
   EXPECT_EQ(escape_first_character_in_keyword(u8"ZYXW"_sv), u8"\\u{5a}YXW");
 }
+
+TEST(test_no_overflow, parser_depth_limit_not_exceeded) {
+  {
+    for (const char8 opening_paren : {u8'(', u8'['}) {
+      string8 opening_parens(parser::stack_limit - 1, opening_paren);
+      padded_string code(opening_parens);
+      spy_visitor v;
+      parser p(&code, &v);
+      bool ok = p.parse_and_visit_module_catching_fatal_parse_errors(v);
+      EXPECT_TRUE(ok);
+    }
+  }
+
+  {
+    string8 left_curly(parser::stack_limit, '{');
+    padded_string code(left_curly);
```

Comment:
Nit: Why are the limits for two tests different (other than `[` vs `{`)?

Reply:
This is because of the way `[` and `{` are parsed, the number of function calls (to `parse_expression`) made for `((` or `[[` is 3 times and for `{{` it is 2 times.

Reply:
That specific different sounds uninteresting to test. I'd merge these test cases into one loop, and test with `parser::stack_limit - 1` repetitions.